src/lib/Libwin/accesinfo.c

```diff
@@ -80,16 +80,18 @@ create_secure_dacl(char *user, ACCESS_MASK mask, SID *owner_sid)
 	int k = 0;
 	int cbAcl = 0;
 	ACL *ndacl = NULL;
-	char logb[LOG_BUF_SIZE] = {'\0' };
+	char logb[LOG_BUF_SIZE] = {'\0' } ;
 
 	rids[0] = DOMAIN_ALIAS_RID_ADMINS;
 	k = getgids(getlogin(), grp, rids);
+	if (k <= 0) {
+		log_eventf(PBSEVENT_ERROR, 0, LOG_DEBUG, __func__, "failed to get group-id");
+	}
 
 	if ((k < _MAX_GROUPS) && (owner_sid != NULL)) {
 		grp[k] = sid_dup(owner_sid);
 		if (grp[k] == NULL) {
-			sprintf(logb, "failed to copy owner_sid");
-			log_err(-1, __func__, logb);
+			log_eventf(PBSEVENT_ERROR, 0, LOG_DEBUG, __func__, "failed to copy owner_sid");
```

Comment:
also no error code in these function failures?

Reply:
No, because that is already handled inside "sid_dup"